### PR TITLE
Re-optimized unswizzle

### PR DESCRIPTION
**Pull request by shinjich**
modified code that optimized for Xenon architecture, and vectorized unity table generators.

### CODE REVIEW

File: XDSP/XDSP.h [hunk 1]
```diff
@@ -93,34 +94,34 @@ namespace XDSP
         using namespace DirectX;
 
         // sign constants for radix-4 butterflies
-        static const XMVECTORF32 vDFT4SignBits1 = { { { 1.0f, -1.0f,  1.0f, -1.0f } } };
```
Comment (walbourn) [inline XDSP/XDSP.h:96]: While not required for Visual C++, the extra braces here are actually needed for C++ conformance.

With clang/LLVM, not having these extra braces results in ``-Wmissing-braces``.

Please put them back for the various ``XMVECTORF32`` cases you removed them from.

**Comment by walbourn** [comment]
Make sure this still runs with the associated **xdsp** test [here](https://github.com/walbourn/directxmathtest/wiki).

The CMakeLists.txt in that project should also allow you to verify it builds without warnings with clang/LLVM.

File: XDSP/XDSP.h (hunk 1 above)
Comment (shinjich) [reply XDSP/XDSP.h:96]: Thank you for reviewing.
I put extra braces back for clang/LLVM and re-committed. At that time, the indent contained tabs, so I committed twice to fix it.

I also tested a combination of each platform/configuration and compiler switch using the directxmathtest/XDSPTest you pointed to.
And I got the following results without warnings in all of them:

FFT4 passed
FFT8 passed
FFT16 passed
FFT (32) passed
FFTUnswizzle passed
FFTPolar passed
Deinterleave passed
Interleave passed
FFTInterleaved passed
IFFTDeinterleaved passed

**Comment by shinjich** [comment]
The updated XDSP adopted XMVectorSinCos for vectorization. I noticed that compiling with clang LLVM would cause an accuracy error of about 1 bit and it would fail compareson on XDSPTest.
#define TESTEPSILON 0.00001f -> 0.0000124f can absorb accuracy errors, but if we need to resolve the accuracy error itself, we will need to use standard library's sinf/cosf instead of XMVoctorSinCos.
Note that this issue does not occurs with VC compiler.

**Comment by walbourn** [comment]
> The updated XDSP adopted XMVectorSinCos for vectorization. I noticed that compiling with clang LLVM would cause an accuracy error of about 1 bit and it would fail compareson on XDSPTest.
> #define TESTEPSILON 0.00001f -> 0.0000124f can absorb accuracy errors, but if we need to resolve the accuracy error itself, we will need to use standard library's sinf/cosf instead of XMVectorSinCos.
> Note that this issue does not occurs with VC compiler.

Feel free to submit a PR for the test as well. It's pretty common to need to adjust epsilon between implementations. You should see what I hit originally moving from Intel to ARM :)

**Comment by shinjich** [comment]
Thanks Chuck,
I'll submit a PR for the test later. As you pointed out, precision will be different between NEON and SSE.

**Review by chrissy35 — APPROVED** [review]

**Review by walbourn — APPROVED** [review]

**Pull request merged**